implementation/src/main/java/io/smallrye/config/SmallRyeConfigBuilder.java

```diff
@@ -150,6 +153,17 @@ public OptionalInt getPriority() {
                 return OptionalInt.of(600);
             }
         }));
+        interceptors.add(new InterceptorWithPriority(new ConfigSourceInterceptorFactory() {
+            @Override
+            public ConfigSourceInterceptor getInterceptor(final ConfigSourceInterceptorContext context) {
+                return new SecretKeysConfigSourceInterceptor(secretKeys);
+            }
+
+            @Override
+            public OptionalInt getPriority() {
+                return OptionalInt.of(-Integer.MAX_VALUE);
```

Comment:
I'm not sure this interceptor should be absolutely first.  It might then miss lookups on keys that are relocated or otherwise manipulated by the user.

I wonder if we should document our interceptor priorities with notes about where user interceptors should be placed.  IIRC some specifications define useful integer constants for users to reference.

Reply:
Sure, we can add some integer constants for the interceptors. That may also help when we add support to interleave interceptors and config sources.